reference/regexp.go

```diff
@@ -96,14 +99,23 @@ var (
 	// that may be part of image names. This is purposely a subset of what is
 	// allowed by DNS to ensure backwards compatibility with Docker image
 	// names. This includes IPv4 addresses on decimal format.
-	domainName = domainNameComponent + anyTimes(`\.`+domainNameComponent)
+	//
+	// TODO(thaJeztah): disambiguate: docker requires domain-name to be either;
+	// - localhost (special case)
+	// - at least one "."
+	// - or a ":port"
+	//
+	// Any other domain is considered a path-element.
+	domainName = domainNameComponent + oneOrMore(`\.`+domainNameComponent)
```

Comment:
This is still to be looked at; I think the new matches are already better than the old ones, but needs some eyes

Reply:
^^ to outline; previously the Regex would handle these as "namespace", and then the code processing the output would mark them as "domain". In this implementation, the Regex already splits them to be a domain (after which the code can do further handling).


(this is why some of the tests were updated, because they tested the Regex "incorrect" behavior)